Fix:empty segments crashed the new version of XTreeGetRecord

GabrieleManduchi opened the pull request:
The crash occurred in Squish section. A check has been added before in order to pass only non empty segments to Squish function.

GabrieleManduchi committed:
Fix:empty segments crashed the new version of XTreeGetRecord

The crash occurred in Squish section. A check has been added before in order to pass only non empty segments to Squish function.

zack-vii commented on xtreeshr/XTreeGetTimedRecord.c:
```diff
@@ -341,7 +341,11 @@ EXPORT int XTreeGetTimedRecord(int inNid, mdsdsc_t *startD,
+    if(resampledXds[currSegIdx].l_length > 0)
+      signals[nonEmptySegIdx++] = (mds_signal_t *)resampledXds[currSegIdx].pointer;
+    else //Keep track that the segment is not considered
+      signalsApd.arsize -= sizeof(mds_signal_t *);
```
shoukd it not better be using the length field of signalsApd instead of sizeof. best would be though to calculate arsize based on nonEmptySegIdx an length after the for loop.

zack-vii replied:
we should add some tests that tackle those kind of edge cases because xtreeshr is used frequently and its a pitty its not tested well. i may be able to contribute. we should add these tests to treeUnitTest where i test the timecontext stuff. maybe there is a lightweight example that would trigger the issue w/o preexisting tree files.

GabrieleManduchi replied:
Ok, removed length update after the loop

zack-vii reviewed (changes requested):
the arsize should be calculated at the end of the loop.
and we need tests. but the can follow in a different pr

GabrieleManduchi committed:
Fix:handle empty segments in XTreeGetTimedRecord

Empty segments may appear after resampling. Now removed before going to squish them
